### ISSUE TITLE
Refuse to install pkgs without specified license when allowUnfree = false

### ISSUE THREAD

**Issue by anjefu** [NixOS/nixpkgs#43231]
## Issue description

It appears there are currently over 2500 packages with no meta.license specified. When allowUnfree = false, they seem to still install without issue. I think nix should probably refuse to install them until their license has been specified, to avoid accidentally installing non-free packages.

**Comment by xeji** [comment]
This would be essentially equivalent to making `licenses.unfree` the default value for `meta.license`.

**Comment by matthewbauer** [comment]
A potential issue with something like that is lots of people use stdenv.mkDerivation for stuff outside of Nixpkgs where they are not as concerned with licensing. It might be better to make sure everything has a license in Nixpkg through something like make-tarball.nix (https://github.com/nixos/nixpkgs/blob/master/pkgs/top-level/make-tarball.nix#L77-L97) & make it block the channels. This would also be useful for making sure ```builtins.parseDrvName``` works correctly.

**Comment by LnL7** [comment]
This doesn't seem like a good idea, there are lot's of meta packages in nixpkgs like buildEnv which don't have (and probably shouldn't have) license information. Anything unfree in nixpkgs that doesn't have a license specified is a problem and should be fixed.

**Comment by matthewbauer** [comment]
> This doesn't seem like a good idea, there are lot's of meta packages in nixpkgs like buildEnv which don't have (and probably shouldn't have) license information

I disagree - everything list in all-packages.nix should have a license. This is what is reported in various tools like ```nix search``, ```nix-env -qa```, and Hydra. It's kind of weird for certain things where there are legitimately multiple licenses but in those cases you can combine them.

**Comment by LnL7** [comment]
I was mostly referring to making `nix-env -i` fail, since that doesn't make sense in a lot of cases. Something like making checkMeta enforce a license for derivations with a src/srcs makes more sense to me. There are things nix all-packages.nix like setup hooks that are just part of nixpkgs, and only enforcing toplevel attributes still hides other potential problems.

**Comment by aanderse** [comment]
@matthewbauer so agreement is that we should just continue to work on getting licenses listed and close this issue?
